### ISSUE TITLE
Modeling table with frozen collections columns in Scala

### ISSUE THREAD

**Issue by AnatoliiStepaniuk** [outworkers/phantom#676]
How can I model a table like this in Scala?
```
create table items_table (
userid int primary key,
items map<int, frozen<list<double>>>
)
```
A line like this (without `frozen` keyword):
`object items extends MapColumn[Int, List[Double]](this)`
results in obvious error `Non-frozen collections are not allowed inside collections: map<int, list<double>>`.
Unfortunately could not find example of using frozen collections in documentation. Is it available in opensource version of Phantom?

**Comment by alexflav23** [comment]
Hi Anatolii,

Thank you for reporting this!

I see, yes it should be OSS And everything, and it should be automated. Would you mind sharing the schema DSL with me? I will have a look, there are a few bugs around this apparently and I'll capture them in one ticket and do a fix today.

Regards,
Flavian

**Comment by AnatoliiStepaniuk** [comment]
Thank you, Flavian!

Hope this is helpful:
I would like to model a table like this:
```
create table items_table (
userid int primary key,
items map<int, frozen<list<double>>>
)
```
And I've tried this code (I should have used **frozen** somewhere here, but not sure how to do that - please advice me on this):
```
import com.outworkers.phantom.CassandraTable
import com.outworkers.phantom.dsl._
import scala.concurrent.Future

case class Items(userId: Int, items: Map[Int, List[Double]])

abstract class ItemsModel extends CassandraTable[ConcreteItemsModel, Items] {

  override def tableName: String = "items_table"

  object userId extends IntColumn(this) with PartitionKey

  object items extends MapColumn[Int, List[Double]](this)

  override def fromRow(r: Row): Items = Items(userId(r), items(r))
}

abstract class ConcreteItemsModel extends ItemsModel with RootConnector {

  def getItemsByUser(userId: Int): Future[List[Map[Int, List[Double]]]] = {
    select(_.items)
      .consistencyLevel_=(ConsistencyLevel.ONE)
      .where(_.userId eqs userId)
      .fetch
  }

  def save(userId: Int, itemsWithFeatures: Map[Int, List[Double]]) = {
    update
      .consistencyLevel_=(ConsistencyLevel.ONE)
      .where(_.userId eqs userId)
      .modify(_.items.putAll(itemsWithFeatures))
      .future()
  }
}
```

**Comment by AnatoliiStepaniuk** [comment]
@alexflav23 Do you mean, that after the fix I should be able to just use ` object items extends MapColumn[Int, List[Double]](this)` without need to put a **frozen** keyword?
Or should I manually describe what exactly I want to be frozen? (in this case, can you please give an example on how it can be used?)

**Comment by alexflav23** [comment]
Hi @AnatoliiStepaniuk Fixed in 2.7.6, should be on Maven Central within 1 hour or so.
